Development to master branch

morganblackwell opened the pull request:
Code to push to master before it is reviewed

morganblackwell committed:
Implemented a basic property and Roll method

morganblackwell committed:
Die rolls duplicated to fix

morganblackwell committed:
Fixed duplicate results

morganblackwell committed:
Testing class implemented

morganblackwell committed:
Infinte loop, XML comments

morganblackwell committed:
XML comments

morganblackwell committed:
XML comments

morganblackwell committed:
XML comments

morganblackwell committed:
Added .ToLower() to input

morganblackwell committed:
Added comments

Xeroas commented on CMP1903_A1_2324/Die.cs:
```diff
@@ -13,11 +13,26 @@ internal class Die
 
-        //Method
+        // Property
+        public int CurrentValue
```
More commenting could be done here to explain what is happening

Xeroas commented on CMP1903_A1_2324/Program.cs:
```diff
@@ -14,6 +14,29 @@ static void Main(string[] args)
+            for (; ; )
+            {
+                Testing testOutputs = new Testing();
+                testOutputs.Main();
```
Game seems to be ran through the testing class, not game class - is this intentional?

Xeroas commented on CMP1903_A1_2324/Testing.cs:
```diff
@@ -15,6 +16,39 @@ internal class Testing
+        /// </summary>
+
         //Method
+        public void Main()
```
Good handling of all potential errors, case type is consistent with C# convention.

Xeroas commented on CMP1903_A1_2324/Testing.cs:
```diff
@@ -15,6 +16,39 @@ internal class Testing
+            int sum = testOutputs.Item4;
+
+            // First dice testing
+            Debug.Assert(roll1 <= 6, "First dice roll is greater than 6");
```
This could be further improved by running on one line using "&&"

Tr0-p replied:
I agree with Jack, the use of and will decrease the amount of repeated lines.

morganblackwell replied:
Decided to keep these lines separate to provide more information about the errors

Xeroas commented on CMP1903_A1_2324/Program.cs:
```diff
@@ -14,6 +14,29 @@ static void Main(string[] args)
+
+
+            // Keep rolling until 'stop' is typed
+            for (; ; )
```
No Syntax errors, code runs as expected

Tr0-p replied:
A use of a indefinite while loop may be a better choice compared to a definite for loop.

Xeroas commented on CMP1903_A1_2324/Game.cs:
```diff
@@ -15,7 +15,30 @@ internal class Game
+        public (int, int, int, int) Main()
+        {
+            // Create 3 dice
+            Die die1 = new Die();
```
Formatted very nicely, very easy to read and understand what is happening

Xeroas reviewed (commented):
Great piece of code, very little that could be improved!

1. Have the requirements been met? Yes
2. Is the code formatted using the Style Guidelines correctly? Yes
3. Is the code easy to read? Yes
4. Are different errors handled correctly? Die rolls are checked to ensure they are within the expected range but no formal (Try block) error handling is implemented, however the code handles unforeseen inputs in the one user input without any faults or breaking.

Tr0-p commented on .idea/vcs.xml:
```diff
@@ -0,0 +1,6 @@
+  <component name="VcsDirectoryMappings">
+    <mapping directory="" vcs="Git" />
+  </component>
+</project>
```
These files inside of inspectionProfiles can be added to Git Ignore.

Tr0-p commented on CMP1903_A1_2324/Die.cs:
```diff
@@ -13,11 +13,26 @@ internal class Die
          */
 
-        //Property
+        /// <summary> returns the currentValue of a randomly rolled 6 sided die. </summary>
```
Good use of XML commenting however some more detail could be added.

Tr0-p commented on CMP1903_A1_2324/Die.cs:
```diff
@@ -13,11 +13,26 @@ internal class Die
-        //Property
+        /// <summary> returns the currentValue of a randomly rolled 6 sided die. </summary>
 
+        // Private variable
```
Redundant Comment. More detail can be added to show what is being stored.

Tr0-p commented on CMP1903_A1_2324/Die.cs:
```diff
@@ -13,11 +13,26 @@ internal class Die
+        // Method
+        // Returns a random integer between 1 and 6
+        public int Roll()
+        {
```
Some Function XML comments can be added to describe what is going on in the function alongside expected returns.

Tr0-p commented on CMP1903_A1_2324/Game.cs:
```diff
@@ -15,7 +15,30 @@ internal class Game
+
 
     }
 }
```
Functionality for Multiple Rolls would be a good addition to increase the capability of the code.

Tr0-p commented on CMP1903_A1_2324/Game.cs:
```diff
@@ -15,7 +15,30 @@ internal class Game
-        //Methods
+        /// <summary>  Creates three die objects, rolls them and outputs the values and their sum. </summary>
+        /// 
+        /// <returns> first die value, second die value, third die value, sum of all dice. </returns>
```
Good use of XML commenting. Returns should be inside the function as it is not run upon instantiation.

Tr0-p reviewed (commented):
Only a few commenting errors which can be fixed alongside some functionality clean-ups, overall a good piece of code!

1. Have the requirements been met? Yes
2. Is the code formatted using the Style Guidelines correctly? Yes
3. Is the code easy to read? Yes
4. Are different errors handled correctly? There is no evidence of error handling pres

morganblackwell committed:
Added user input for dice rolls

The user can choose how many times the dice are rolled, utilised exception handling

morganblackwell committed:
Added, edited and rearranged comments

Added some comments and moved some around to improve readability
